### ISSUE TITLE
[SB2] Virtual Keyboard doesn't come up on Browsers

### ISSUE THREAD

**Issue by whatevergit** [linux-surface/linux-surface#819]
Hello,

my first Github-Post! Please tell me if information is missing.

### The Problem:
When in tablet-mode or tablet-mode with second half attached: the virtual keyboard doesn't pop up when clicking on a text field or the address bar in a Browser like Firefox or Chromium. 
The virtual keyboard does open in other applications, like LibreOffice Writer.

Is there a solution for me to type text in a Browser when in tablet mode?

### Environment

- Hardware model: Surface Book 2
- Kernel version: 5.17.5-surface
- Distribution: Debian 11
- Software: Firefox 91.9.0esr (64-bit); Chromium 101.0.4951.41 (Official Build) 

Thank you for all the work you put in!

**Comment by flying-foss** [comment]
Same here at the surface go 3. 

It seems to be a problem with wayland. The onscreen keyboard shows up in firefox by using the x server. But it is so damn slow that this is no real alternative. I hope this will get fixed soon. Having no onscreen keyboard is the biggest problem for not using my surface as a tablet-only machine. If someone has any idea or workaround.. please let us know. 

Thank you :)

**Comment by flying-foss** [comment]
I found a workaround that works:

https://www.reddit.com/r/gnome/comments/orgket/onscreen_keyboard_only_works_for_gnome/

install the gnome extension custom-hot-corners-extended:
https://github.com/G-dH/custom-hot-corners-extended/tree/gdh

you can choose an active corner and add the "toggle onscreen keyboard" feature.

it can be found in the options of the extension in the dropdown  menu for the active corner in the section "universal access".

The feature works with the mouse, but not with the touchscreen. To enable the touchscreen "gesture" you can go in the section "options" on the left of the extension options and active the toggle "use fallback hot corner triggers"

now you should be able to touch the corner with the finger and the keyboard should pop up- even in firefox etc.

Enjoy!

**Comment by JulianHeuser** [comment]
I had this issue on a SGO3 back when I used Ubuntu 22.04, I switched to fedora 36 (gnome) and the on-screen keyboard works great in tablet mode now. Maybe it's an issue with snaps but since you're using debian I don't think that's the case.

**Comment by vitalygashkov** [comment]
Same issue on SLS with Ubuntu 22.10 (Wayland).

**Comment by qzed** [comment]
IIRC @StollD explained it nicely in another issue or on our Matrix chat (better than I probably can): The issue is with the desktop environment (which is generally responsible for automatically bringing up the keyboard) not being able to recognize when you select an input field in browsers (or some other applications that don't make use of the common GUI frameworks, I'm not even sure if e.g. QT/KDE apps work properly on Gnome).

So in short, we're missing a common standard in Linux user-space to signal when an input field has been selected, so that the desktop environment can perform actions based on this. I think there might be something in the works regarding that (I think some freedesktop/XDG portal spec?), but I don't really remember.

**Comment by JulianHeuser** [comment]
If that's the case then why does it work fine for me in fedora? When it's in tablet mode, input fields in firefox bring up the on-screen keyboard.

**Comment by qzed** [comment]
So I checked again a bit and what I alluded to before is `zwp_text_input_manager_v3`. Not all applications support it (AFAICT chromium only supports v1 but not v3, and Gnome supports only v3, and v1 and v3 are not compatible). I'm assuming somewhat recent FIrefox versions do support that, but I'm not sure how well that works across snap/flatpack/appimage boundaries.

So it's very much dependent on the software you're running, including both the desktop environment, the application itself, and possibly also the way it's packaged.

**Comment by NexGen-3D-Printing** [comment]
Still an issue today on nearly every Linux OS and every browser except for Firefox.
